### PR TITLE
Unify settings names

### PR DESCRIPTION
**Pull request by Asnelchristian**
related to https://github.com/coala-analyzer/coala-bears/issues/460

### CODE REVIEW

File: bears/coffee_script/CoffeeLintBear.py [hunk 1]
```diff
@@ -207,8 +207,8 @@ class Foo
         :param consistent_line_endings_style:
             The option to ``line_endings``, its value is either ``unix`` or
             ``windows``.
-        :param no_this:
-            Prohibits ``this``. Use ``@`` instead.
+        :param allow_this_statements:
+            Allows the use of ``this``.
```
Comment (sils1297) [inline bears/coffee_script/CoffeeLintBear.py]: I feel like the hint wiht the use `@` thing is important

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/638/commits/c1a05a86c854e7fcd1adf6c5b4d3411b33c47f14

File: bears/coffee_script/CoffeeLintBear.py [hunk 2]
```diff
@@ -211,8 +211,8 @@ class Foo
             Allows the use of ``this``.
         :param allow_increment:
             Allows the use of increment and decrement arithmetic operators.
-        :param no_empty_parameter_list:
-            Prohibits empty parameter lists in function definitions.
+        :param allow_empty_parameter_list:
```
Comment (sils1297) [inline bears/coffee_script/CoffeeLintBear.py]: maybe rather do like `allow_no_parameters`?

File: bears/coffee_script/CoffeeLintBear.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def generate_config(filename, file,
                         allow_this_statements: bool=True,
                         allow_increment: bool=True,
                         allow_empty_parameter_list: bool=True,
-                        no_empty_functions: bool=False,
+                        allow_empty_functions: bool=True,
```
Comment (sils1297) [inline bears/coffee_script/CoffeeLintBear.py]: can you make another commit changing the default to False? I think we should warn here. FWIW we should go for sane defaults rather than just taking the linters defaults.

Comment (sils1297) [inline bears/coffee_script/CoffeeLintBear.py]: IMO

**Comment by sils1297** [comment]
have the commits below been acked earlier?

**Comment by Asnelchristian** [comment]
no, this is weird

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/638/commits/54daf64bd82333f21cdb7e7b62e7ec8d99e554f3

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/638/commits/215993710eea8646afdb393bd3c166cdc6ecf13c https://github.com/coala-analyzer/coala-bears/pull/638/commits/ec53af381365009a68a8f895fc2721a2fe0ca820 https://github.com/coala-analyzer/coala-bears/pull/638/commits/36e974d6d7503b81a726657d026787d80a4a503f

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/638/commits/6312c373a1fde798f9f7b94f28ec69585e715843

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/638/commits/87b38db8a1cddd9f151d7e58085d21fca6515065 https://github.com/coala-analyzer/coala-bears/pull/638/commits/135cc13a72e4af83e3db470f40c6a4e998195504 https://github.com/coala-analyzer/coala-bears/pull/638/commits/21e3ac8ac6751db8c6436e88babdaa1675bfa3fa https://github.com/coala-analyzer/coala-bears/pull/638/commits/225abe17dc4714b26e60828cfcd8ec62c7122bf8 https://github.com/coala-analyzer/coala-bears/pull/638/commits/c144241d71aae2c296d2b1dc97d6230d32554558 https://github.com/coala-analyzer/coala-bears/pull/638/commits/f404bdb9422ac8f95d44fcc22233320877e24343 https://github.com/coala-analyzer/coala-bears/pull/638/commits/fbd269ca0d37eaf20f12c126b3b3f22e24a21ac5

**Comment by sils1297** [comment]
@Asnelchristian rebase and merge. FWIW try to keep your commit messages a bit more precise, "Change default value" is pretty abstract, one could rather use "Allow ... by default"

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/638/commits/0a52d53383ca254b660337dbfa1b0ce398894f28 https://github.com/coala-analyzer/coala-bears/pull/638/commits/7d43eb6834b904aed2f8f598d73c42720d45fbfa https://github.com/coala-analyzer/coala-bears/pull/638/commits/a85087caee22a4541a2a6fac5a03f2a6e5330f3f https://github.com/coala-analyzer/coala-bears/pull/638/commits/aaad14d10148906ab552c53a79a5c070dd45be97 https://github.com/coala-analyzer/coala-bears/pull/638/commits/bc703b63f784559828aad57d71afd3491df969f6 https://github.com/coala-analyzer/coala-bears/pull/638/commits/53d45d711a0f1578b4e4042e0c37455da84e2f93

**Comment by Asnelchristian** [comment]
@rultor merge
